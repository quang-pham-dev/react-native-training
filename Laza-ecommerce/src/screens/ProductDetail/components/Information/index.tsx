import React, { memo } from 'react';
import { View, Text } from 'react-native';

//Types
import { IProductDetailItemProps } from 'types/screens/ProductDetail';

// Styles
import styles from './styles';

const Information = ({ title, type, price }: IProductDetailItemProps) => {
  return (
    <View style={styles.productInfoContainer}>
      <View style={styles.productInfoWrapper}>
        {Boolean(title && type) && (
          <>
            <Text style={styles.textTitle}>{title}</Text>
            <Text style={styles.textValue}>{type}</Text>
          </>
        )}
      </View>

      <View style={styles.productInfoWrapper}>
        <Text style={styles.textTitle}>Price</Text>
        {Boolean(price) && <Text style={styles.textValue}>{`$${price}`}</Text>}
      </View>
    </View>
  );
};

export default memo(Information);
