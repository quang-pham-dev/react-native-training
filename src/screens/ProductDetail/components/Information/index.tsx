import React, {memo} from 'react'

// Libs
import isEqual from 'react-fast-compare'
import {vs} from 'react-native-size-matters/extend'

// Styles
import PStyled from '@components/Paragraph/P.styles'
import FlexStyled from '@components/Flex/Flex.styles'

// Constants
import {PARAGRAPH_TYPE} from '@constants'
import ViewStyled from '@components/View/View.styles'

// Themes
import {Metrics} from '@themes'

type InformationProps = {
  name?: string
  type?: string
  price?: number
}

const Information = ({name, type, price}: InformationProps) => {
  return (
    <ViewStyled.Custom pVertical={vs(Metrics.padding.medium)}>
      <FlexStyled.FlexSpaceBetween pBottom={vs(8)}>
        <PStyled.Base type={PARAGRAPH_TYPE.PRODUCT_DETAIL_TITLE}>
          {name}
        </PStyled.Base>
        <PStyled.Base type={PARAGRAPH_TYPE.PRODUCT_DETAIL_TITLE}>
          Price
        </PStyled.Base>
      </FlexStyled.FlexSpaceBetween>

      <FlexStyled.FlexSpaceBetween>
        <PStyled.Base type={PARAGRAPH_TYPE.PRODUCT_DETAIL_CONTENT_TITLE}>
          {type}
        </PStyled.Base>
        <PStyled.Base>{`$${price}`}</PStyled.Base>
      </FlexStyled.FlexSpaceBetween>
    </ViewStyled.Custom>
  )
}

export default memo(Information, isEqual)
