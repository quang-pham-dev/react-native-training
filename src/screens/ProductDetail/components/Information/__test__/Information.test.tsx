import React from 'react'

// LIBS
import renderer from 'react-test-renderer'

// Components
import Information from '@screens/ProductDetail/components/Information'

// Mock data
import {product} from '@constants'

describe('Product detail information', () => {
  const props: any = {
    title: product.name,
    type: product.type,
    price: product.price,
  }

  const tree = renderer.create(<Information {...props} />)

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot()
  })
})
