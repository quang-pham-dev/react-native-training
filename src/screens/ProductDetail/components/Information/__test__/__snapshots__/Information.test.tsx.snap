// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product detail information should render correctly 1`] = `
<View
  pVertical={15}
  style={
    Array [
      Object {
        "paddingVertical": 15,
        "width": "100%",
      },
    ]
  }
>
  <View
    pBottom={8}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#8F959E",
            "fontFamily": "Inter-Regular",
            "fontSize": 13,
            "lineHeight": 14.3,
            "textAlign": "left",
          },
        ]
      }
      type="productDetailTitle"
    />
    <Text
      style={
        Array [
          Object {
            "color": "#8F959E",
            "fontFamily": "Inter-Regular",
            "fontSize": 13,
            "lineHeight": 14.3,
            "textAlign": "left",
          },
        ]
      }
      type="productDetailTitle"
    >
      Price
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1D1E20",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 22,
            "lineHeight": 24.2,
            "textAlign": "left",
          },
        ]
      }
      type="productDetailContentTitle"
    >
      Cotton
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "left",
          },
        ]
      }
    >
      $100
    </Text>
  </View>
</View>
`;
